### DISCUSSION TITLE
Campo Date

### DISCUSSION THREAD

**Discussion by valdirdill** [viniciussanchez/RESTRequest4Delphi#200] [General]
Bom dia,
Numa rotina qualquer, trago um jsonObject do servidor com o RESTRequest4Delphi e, no lado cliente, alimento um TMDMemTable direto na requisição (.Adapters(TDataSetSerializeAdapter.New(TDMemTable1)).
Um problema que estou enfrentando é que nos campos tipo ftDate, ele fica como ftString. Aí depois quando vai manipular esse campo/dado, dá problemas.
Tem alguma configuração para que o serialize mantenha os dataType originais da tabela?

Obrigado

**Comment by viniciussanchez** [comment]
Bom dia... quando os fields são criados de forma automática, é criado com base na leitura do JSON. JSON não trabalha com campos Data. JSON só reconhece campos Numericos, Strings e Boolean. Caso queira que ele respeite os tipos de dados que você quer, é necessário que você crie os fields de forma manual no seu DataSet. Assim ele vai respeitar os fields que já estão criados.

Quando a tua pergunda: Tem alguma configuração para que o serialize mantenha os dataType originais da tabela?
Não tem, até porque você está usando um TFDMemTable, que por sua vez, nem acesso a banco de dados possui.
